packages/react-components/src/components/time-sync/viewportAdapter.tsx

```diff
@@ -119,3 +119,31 @@ export const viewportToDateRange = (viewport?: Viewport): DateRangePickerProps.V
 
   return null;
 };
+
+export const getNewViewportStartDate = (value: DateRangePickerProps.RelativeValue): Date => {
```

Comment:
nit: the name doesn't capture how the function get the new start date, maybe like `getBackwardRelativeStartDate`? 🤔